src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -107,12 +106,45 @@ AttributeSyntax attribute
 		return returnType;
 	}
 
+	private TypeSyntax TransformType( TypeSyntax typeSynt ) {
+		var returnTypeInfo = Model.GetTypeInfo( typeSynt, Token );
+
+		if( returnTypeInfo.Type == null ) {
+			GeneratorError( typeSynt.GetLocation(), "Couldn't resolve type" );
+			return typeSynt;
+		}
+
+		if( returnTypeInfo.Type.ContainingNamespace.ToString() == "System.Threading.Tasks" ) {
+			switch( returnTypeInfo.Type.MetadataName ) {
+				case "Task":
+					return SyntaxFactory.ParseTypeName( "void" )
+						.WithTriviaFrom( typeSynt );
+				case "Task`1":
+					return ( (GenericNameSyntax)typeSynt )
+						.TypeArgumentList.Arguments.First()
+						.WithTriviaFrom( typeSynt );
+
+				default:
+					GeneratorError(
+						typeSynt.GetLocation(),
+						$"Unexpected Task type: {returnTypeInfo.Type.MetadataName}"
+					);
+					return typeSynt;
+			}
+		}
+
+		return typeSynt;
+	}

```

Comment:
At a glance this looks the same as `TransformReturnType`?
If they are the same, is there a reason for having both? Could we just rename `TransformReturnType` to `TransformType` if it reads better?

Reply:
I made TransformType so that ReportDiagnostic( Diagnostics.NonTaskReturnType, returnType.GetLocation() ); would not be reported for non-return types of a function. Should I combine them and add an optional param?

Reply:
Ah ya I see. That would make sense to me!